src/kem/kyber/libjade_kyber768_ref/api.c

```diff
@@ -0,0 +1,19 @@
+#include <oqs/rand.h>
+#include "api.h"
+
+int libjade_kyber768_ref_keypair(uint8_t *public_key, uint8_t *secret_key) {
+    uint8_t keypair_coins[JADE_KEM_kyber_kyber768_amd64_ref_KEYPAIRCOINBYTES];
+    OQS_randombytes(keypair_coins, JADE_KEM_kyber_kyber768_amd64_ref_KEYPAIRCOINBYTES/2);
+    OQS_randombytes((uint8_t *)(keypair_coins + (JADE_KEM_kyber_kyber768_amd64_ref_KEYPAIRCOINBYTES/2)), JADE_KEM_kyber_kyber768_amd64_ref_KEYPAIRCOINBYTES/2);
```

Comment:
Answering on @praveksharma's behalf: since the NIST DRBG uses a counter, the sequences of calls needs to be duplicated to not screw up the KATs.

Reply:
Thanks, but still not sure why generating the randombytes in one go should lead to different bytes. AFAIK, the counter in the DRBG is incremented after each 16 byte block and the state is kept after each call. As long as one doesn't truncate generated bytes (which doesn't seem the case here), there shouldn't be a difference.

Reply:
It does seem counterintuitive, but I've heard it from a few people, most recently from @tfaoliveira, who I think understands this subtle problem.

Reply:
Looking at the source code, I believe it's not because of the counter (as I had thought) but because the AES key is updated after each `randombytes` call: https://github.com/open-quantum-safe/liboqs/blob/a5ec23cf19763d36a558b8358345823ae45d57e5/src/common/rand/rand_nist.c#L118